geode-core/src/main/java/org/apache/geode/management/DiskBackupStatus.java

```diff
@@ -63,7 +68,37 @@ public void setBackedUpDiskStores(Map<String, String[]> backedUpDiskStores) {
    * Sets the list of directories for the disk stores that were off-line at the time the backup
    * occurred.
    */
-  public void setOfflineDiskStores(String[] offlineDiskStores) {
-    this.offlineDiskStores = offlineDiskStores;
+  @SuppressWarnings("unused")
+  public void setOfflineDiskStores(String[] offLineDiskStores) {
+    this.offlineDiskStores = offLineDiskStores;
+  }
+
+  /**
+   * Sets the map of member names/IDs and the {@link PersistentID} of the disk stores that were
+   * backed up.
+   */
+  public void generateBackedUpDiskStores(
```

Comment:
This is an external, customer api. Would they call "generateBackedUpDiskStores"?

Reply:
The user of this method and the other one in this class you commented on are both used from outside this package, which necessitates that they be public. There is actually no reason that this class is exposed in the API. Maybe the best approach is the convert it to an interface and move the implementation elsewhere?

Reply:
It looks like DiskBackupStatus already exposed apis it should not of. It should just have gettor methods. It is part of the external api exposed by a method on DistributedSystemMXBean.

I think the best solution would be to do what you suggest; make this class an interface; the interface should only have the gettor methods; remove the two settor methods from the interface and move them to the internal impl. Your new methods would also go on this method. Any customer currently calling these settors should not be so I think this would be an okay non-backward compatible change.

Reply:
Actually DiskBackupStatus may need to be a simple bean because it is used as a jmx composite type. So it might be best to just not change it at all move these generate methods into an internal calls that can just call this guy's settors. Kirk Lund would be a good person to talk about this to.

Reply:
This class is defining a composite data type for the DistributedSystemMXBean. Any changes need to remain faithful to the conventions of JMX open data types for MXBean.